js/bybit.js

```diff
@@ -2233,15 +2273,27 @@ module.exports = class bybit extends Exchange {
          * @method
          * @name bybit#fetchBalance
          * @description query for balance and get the amount of funds available for trading or funds locked in orders
+         * @see https://bybit-exchange.github.io/docs/spot/#t-balance spot
+         * @see https://bybit-exchange.github.io/docs/spot/#t-queryaccountinfo cross margin
+         * @see https://bybit-exchange.github.io/docs/inverse/#t-balance USDT/inverse perpetual & inverse futures
+         * @see https://bybit-exchange.github.io/docs/usdc/perpetual/#t-usdcaccountinfo USDC
          * @param {object} params extra parameters specific to the bybit api endpoint
          * @returns {object} a [balance structure]{@link https://docs.ccxt.com/en/latest/manual.html?#balance-structure}
          */
         const request = {};
         let type = undefined;
+        let marginMode = undefined;
         [ type, params ] = this.handleMarketTypeAndParams ('fetchBalance', undefined, params);
+        [ marginMode, params ] = this.handleMarginModeAndParams ('fetchBalance', params);
         let method = undefined;
         if (type === 'spot') {
-            method = 'privateGetSpotV1Account';
+            if (marginMode === 'cross') {
+                method = 'privateGetSpotV1CrossMarginAccountsBalance';
+            } else if (marginMode === 'isolated') {
+                throw new BadRequest (this.id + ' fetchBalance marginMode must be cross if assigned');
```

Comment:
I would suggest replacing `BadRequest` with `NotSupported` and probably changing the message a bit to something like "fetchBalance() does not support isolated margin...." . Currently I don't find it much clear